test/test_file_utils.py

```diff
@@ -212,3 +213,194 @@ def test_guess_project_dir(tmp_path: Path) -> None:
     with cwd(str(tmp_path)):
         result = guess_project_dir(None)
         assert result == str(tmp_path)
+
+
+BASIC_PLAYBOOK = """
+- name: "playbook"
+  tasks:
+    - name: hello
+      debug:
+        msg: 'world'
+"""
+
+
+@pytest.fixture
+def tmp_updated_lintable(
+    tmp_path: Path, path: str, content: str, updated_content: str
+) -> Lintable:
+    """Create a temp file Lintable with a content update that is not on disk."""
+    lintable = Lintable(tmp_path / path, content)
+    with lintable.path.open("w", encoding="utf-8") as f:
+        f.write(content)
+    # move mtime to a time in the past to avoid race conditions in the test
+    mtime = time.time() - 60 * 60  # 1hr ago
+    os.utime(str(lintable.path), (mtime, mtime))
+    lintable.content = updated_content
+    return lintable
+
+
+@pytest.mark.parametrize(
+    ("path", "content", "updated_content", "updated"),
+    (
+        pytest.param(
+            "no_change.yaml", BASIC_PLAYBOOK, BASIC_PLAYBOOK, False, id="no_change"
+        ),
+        pytest.param(
+            "quotes.yaml",
+            BASIC_PLAYBOOK,
+            BASIC_PLAYBOOK.replace('"', "'"),
+            True,
+            id="updated_quotes",
+        ),
+        pytest.param(
+            "shorten.yaml", BASIC_PLAYBOOK, "# short file\n", True, id="shorten_file"
+        ),
+    ),
+)
+def test_lintable_updated(
+    path: str, content: str, updated_content: str, updated: bool
+) -> None:
+    """Validate ``Lintable.updated`` when setting ``Lintable.content``."""
+    lintable = Lintable(path, content)
+
+    assert lintable.content == content
+
+    lintable.content = updated_content
+
+    assert lintable.content == updated_content
+
+    assert lintable.updated is updated
+
+
+@pytest.mark.parametrize(
+    "updated_content", ((None,), (b"bytes",)), ids=("none", "bytes")
+)
+def test_lintable_content_setter_with_bad_types(updated_content: Any) -> None:
+    """Validate ``Lintable.updated`` when setting ``Lintable.content``."""
+    lintable = Lintable("bad_type.yaml", BASIC_PLAYBOOK)
+    assert lintable.content == BASIC_PLAYBOOK
+
+    with pytest.raises(TypeError):
```

Comment:
It is strongly recommended to always use the `match=` argument for all `pytest.raises()`/`pytest.warns()`/`pytest.deprecated()` uses to prevent accidental illegitimate passes.

Reply:
Doing that feels like relying on implementation details. Had I done that with the old implementation, I would have had to modify it with this implementation because the TypeError was being raised in my code. So, I don't like using the `match=` argument.

This test is not checking for a message. It is checking that it raises a TypeError for non-str things. I don't care why or what message is used, so adding a `match=` here is misleading about my intents when writing the code.

Reply:
Imagine you refactor that code and make a mistake in a function signature you call inside of the setter or down the stack. It will cause a `TypeError` and the test will still pass with a bug merged in 🤷‍♂️